src/modules/ens/reducer.ts

```diff
@@ -0,0 +1,89 @@
+import { ENSData } from './types'
+import { LoadingState, loadingReducer } from 'decentraland-dapps/dist/modules/loading/reducer'
+import {
+  GetENSRequestAction,
+  GetENSSuccessAction,
+  GetENSFailureAction,
+  GET_ENS_REQUEST,
+  GET_ENS_SUCCESS,
+  GET_ENS_FAILURE,
+  SetENSRequestAction,
+  SetENSSuccessAction,
+  SetENSFailureAction,
+  SET_ENS_REQUEST,
+  SET_ENS_SUCCESS,
+  SET_ENS_FAILURE,
+} from './actions'
+
+export type ENSState = {
+  data: Record<string, ENSData>
+  loading: LoadingState
+  error: string | null
+}
+
+const INITIAL_STATE: ENSState = {
+  data: {},
+  loading: [],
+  error: null,
+}
+
+export type ENSReducerAction = GetENSRequestAction | GetENSSuccessAction | GetENSFailureAction |
+                                  SetENSRequestAction | SetENSSuccessAction | SetENSFailureAction
+
+export function ensReducer(state: ENSState = INITIAL_STATE, action: ENSReducerAction): ENSState {
+  switch (action.type) {
+    case GET_ENS_REQUEST: {
+      return {
+        ...state,
+        loading: loadingReducer(state.loading, action),
+      }
+    }
+    case GET_ENS_SUCCESS: {
+      const { ens, data } = action.payload
+      
+      return {
+        ...state,
+        loading: loadingReducer(state.loading, action),
+        data: {
+          [ens] : data
+        }
+      }
+    }
+    case GET_ENS_FAILURE: {
+      const { error } = action.payload
+      return {
+        ...state,
+        loading: loadingReducer(state.loading, action),
+        error
+      }
+    }
+    case SET_ENS_REQUEST: {
+      return {
+        ...state,
+        loading: loadingReducer(state.loading, action),
+      }
+    }
+    case SET_ENS_SUCCESS: {
+      const { ens, data } = action.payload
+      
+      return {
+        ...state,
+        loading: loadingReducer(state.loading, action),
+        data: {
+          [ens] : data
```

Comment:
Careful! I think you're missing the rest of the `state.data` object here, like:
```ts
data: {
  ...state.data,
  [ens] : data
}
```

if you don't do that, each time a new `ENS` succeeds it'll overwrite all other `ENS`es

*Same thing for line 48 above!*

Reply:
done! thanks!